app/models/post.rb

```diff
@@ -61,6 +61,7 @@ class Post < ActiveRecord::Base
   belongs_to :image_upload, class_name: "Upload"
 
   has_many :post_hotlinked_media, dependent: :destroy, class_name: "PostHotlinkedMedia"
+  has_many :reviewables, as: :target, dependent: :destroy
```

Comment:
Using `has_many` and not `has_one` here as a post can be associated to multiple reviewables. For example, a moderated post will be associated to a `ReviewableQueuedPost` and then a user could flag it, creating a `ReviewableFlaggedPost`.